src/main/java/com/epam/task/third/logic/ConeCalculator.java

```diff
@@ -0,0 +1,101 @@
+package com.epam.task.third.logic;
+
+import com.epam.task.third.enteties.Circle;
+import com.epam.task.third.enteties.Cone;
+import com.epam.task.third.enteties.Point;
+
+public class ConeCalculator {
+
+    final int DEGREE = 2;
+
+    public boolean isCorrectRadius(double radius) {
+        return radius > 0;
+    }
+
+    public boolean isCone(Cone cone) {
+        Circle circle = cone.getCircle();
+        if (!isCorrectRadius(circle.getRadius())) {
+            return false;
+        }
+        Point pointOfApexCone = cone.getApexOfCone();
+        Point pointOfCenter = circle.getCenterOfCircle();
+
+        if ((pointOfCenter.getX() == pointOfApexCone.getX()) && (pointOfCenter.getY() == pointOfApexCone.getY())
+                && (pointOfCenter.getZ() != pointOfApexCone.getZ())) {
+            return true;
+        } else if ((pointOfCenter.getX() == pointOfApexCone.getX()) && (pointOfCenter.getZ() == pointOfApexCone.getZ())
+                && (pointOfCenter.getY() != pointOfApexCone.getY())) {
+            return true;
+        } else
+            return (pointOfCenter.getY() == pointOfApexCone.getY()) && (pointOfCenter.getZ() == pointOfApexCone.getZ())
+                    && (pointOfCenter.getX() != pointOfApexCone.getX());
+    }
+
+
+    public double separateCone(Cone cone, double lengthOfSeparationCone) {
+        if (lengthOfSeparationCone == 0) {
```

Comment:
looks a bit strage, if you treat 0 differently why you don't treat < 0, > 1 dirrirently